x/mint/types/minter.go

```diff
@@ -30,13 +33,29 @@ func ValidateMinter(minter Minter) error {
 	return nil
 }
 
-func (m Minter) NextEpochProvisions(params Params) sdk.Dec {
-	return m.EpochProvisions.Mul(params.ReductionFactor)
-}
-
 // EpochProvision returns the provisions for a block based on the epoch
 // provisions rate.
 func (m Minter) EpochProvision(params Params) sdk.Coin {
 	provisionAmt := m.EpochProvisions
 	return sdk.NewCoin(params.MintDenom, provisionAmt.TruncateInt())
 }
+
+/*	Returns ScheduledRelease if the date of the block matches the scheduled release date.
+ *	You may only schedule one release of tokens on each day, the date must be in
+ *  types.TokenReleaseDateFormat.
+ */
+func GetScheduledTokenRelease(
```

Comment:
Somewhat inefficient as we need to loop through the entire list of release dates, but the list will not grow beyond 10-20 years the list should be bounded by O(10-20). 

Happy to hear any suggestions for how to make this more efficient, one idea I had was to store the index of the previously applied list in the chain but that would be very fragile to param-changes in the future